python/ray/serve/tests/test_controller.py

```diff
@@ -49,6 +52,39 @@ def test(_):
     assert start_time_ms_1 == start_time_ms_2
 
 
+def test_deploy_app_custom_exception(serve_instance):
+    """Check that controller doesn't deserialize an exception from deploy_app."""
+
+    client = serve_instance
+    controller = serve.context.get_global_client()._controller
+
+    config = {
+        "applications": [
+            {
+                "name": "broken_app",
+                "route_prefix": "/broken",
+                "import_path": "broken_dag:app",
+                "runtime_env": {
+                    "working_dir": (
+                        "https://github.com/ray-project/test_dag/"
+                        "archive/e552be913ffb7fb5e36b1e63d97f5c354d45e219.zip"
+                    ),
```

Comment:
let's use a local file instead please to avoid an external dependency

you can import from this file as `from ray.serve.tests.test_controller import my_app`

you can check it isn't serialized by raising an exception in `__reduce__` or by including another non-serializable object as an attribute (and also may as well verify it isn't serializable using `ray.cloudpickle` in the test case)

Reply:
Sounds good, I'll move it to the Ray repo. I can't put it in `test_controller.py` though, since I need to raise the `Exception` in the file itself (outside the deployment), in order for the error to get propagated back to the user.

I'll create a file in `test_config_files` and import it from there.